codersdk/workspaces.go

```diff
@@ -518,6 +518,8 @@ type WorkspaceFilter struct {
 	Limit int `json:"limit,omitempty" typescript:"-"`
 	// FilterQuery supports a raw filter query string
 	FilterQuery string `json:"q,omitempty"`
+	// Shared is a whether the workspace is shared with any users or groups
+	Shared bool `json:"shared" typescript:"-"`
```

Comment:
1. `WorkspaceFilter` has an `asRequestOption` method that also needs to be updated. that's where the `q` param actually gets serialized.

2. this field probably needs to be `*bool` so that you can set `shared:false` as distinct from "no `shared:` filter"